### PR TITLE
Makes gas visuals send less assets

### PR DESCRIPTION
**Pull request by ninjanomnom**
Discord_2023-10-12_13-21-06

~~I'll see how this works out for the downstream, if it improves the situation for them I'll check the profiler to see how much it affects performance.~~

This reduces assets that need to be sent to the client, how much is hard to tell but a downstream that was having issues due to it got reduced issues when testing with this. Server side costs are roughly equivalent now that I switched to using a straight color in the color matrix filter.

### CODE REVIEW

**Comment by MTandi** [comment]
Is this not an issue for chemical overlays?

**Comment by MTandi** [comment]
I can guess that the issue is mostly due to the red (scrubbing) loop constantly sucking something on the station and getting empty. Maybe if you make the overlay appear only when the pressure is above certain small threshold in moles or kPa, it will be resolved.

**Comment by ninjanomnom** [comment]
That is one method I could try but I want to see how well this works first. Pipe gas visuals change color quickly even at low pressures and I'd rather not cut off the bottom end.

**Comment by MTandi** [comment]
Another guess is that it's still drawn for all pipes below the floor tiles, even when they're not visible on the client.
And since the red loop is station-wide, even a single scrubber sucking something causes the entire pipe net to blink for everyone on the station.

**Comment by ninjanomnom** [comment]
So it seems this does work to some extent, now I'll just need to make this perform a bit better and see the final profiler results

**Comment by LemonInTheDark** [comment]
is this unworked on or do you want it reviewed

**Comment by ninjanomnom** [comment]
This is finished and ready

File: code/modules/atmospherics/machinery/datum_pipeline.dm [hunk 1]
```diff
@@ -355,3 +353,24 @@
...
+	color_filter = filters[filters.len]
+	if(current_color)
+		animate(color_filter, color=current_color, time=5)
+
+/obj/effect/abstract/gas_visual/proc/ChangeColor(new_color)
+	current_color = new_color
+	if(isnull(color_filter))
+		// Called before init
+		return
+	animate(color_filter, time=5, color=new_color)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/datum_pipeline.dm:376]: tested? color filters animate weird sometimes

File: code/__HELPERS/colors.dm [hunk 2]
```diff
@@ -17,6 +17,18 @@
 
 	return final_color
 
+/// Given a color in the format of "#RRGGBB" or "#RRGGBBAA", gives back a 4 entry list with the number values of each
+/proc/split_color(color)
```
Comment (LemonInTheDark) [inline code/__HELPERS/colors.dm:21]: why was this added if it's unused? helper?

**Comment by LemonInTheDark** [comment]
so it's not assets but more unique appearances? how much does this impact clientside rendering?

File: code/modules/atmospherics/machinery/datum_pipeline.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/modules/atmospherics/machinery/datum_pipeline.dm:376]: Yeah I tested and it seemed to be fine just giving it a color instead of a matrix

File: code/__HELPERS/colors.dm (hunk 2 above)
Comment (ninjanomnom) [reply code/__HELPERS/colors.dm:21]: Oh, yeah I ended up not needing it but it's still generic and potentially useful so I left it

**Comment by ninjanomnom** [comment]
Clientside I didnt notice any problems and I didnt get any reports that this caused problem on the downstream that needed the fix for any clients. It's hard to tell though. I suspect though that filters get sent to clients as a few number changes, whereas changing the color var probably sends an entire image.

**Comment by LemonInTheDark** [comment]
IG we'll see then
